### PR TITLE
ListBox nested classes cleanup

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

- Automated changes, nothing functional

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3574)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs [hunk 1]
```diff
@@ -240,6 +238,12 @@ internal void AddRangeInternal(ICollection items)
                 }
             }
 
+            object? IList.this[int index]
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:241]: This an below implicit implementations are added to keep the analyzer happy - `IList` signatures accept nullable objects, whilst our implementation doesn't.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs [hunk 2]
```diff
@@ -389,6 +397,8 @@ public void Insert(int index, object item)
                 _owner.UpdateHorizontalExtent();
             }
 
+            void IList.Insert(int index, object? item) => Insert(index, item!);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:400]: We should check for null here if we're claiming that we never let one in.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs [hunk 3]
```diff
@@ -323,6 +329,8 @@ public int IndexOf(object value)
                 return InnerArray.IndexOf(value, 0);
             }
 
+            int IList.IndexOf(object? value) => IndexOf(value!);
```
Comment (JeremyKuhne) [inline src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:332]: Can you just return `-1` or throw here if we get passed a null.

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:332]: Same for contains above

Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:332]: Our implementation `IndexOf` is checking for `null` and throws.

File: src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs (hunk 2 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/ListBox.ObjectCollection.cs:400]: Our implementation of `Insert` performs the check and will throw, if `item is null`.
The same is true for `IList.Contains` - the argument will be asserted

**Comment by RussKie** [comment]
@vladimir-krestov @Ryuugamine FYI

**Pull request merged**